server/instances.go

```diff
@@ -151,6 +151,29 @@ func (p *Plugin) InstallInstance(instance Instance) error {
 				}
 			}
 
+			if instance.Common().Type == CloudOAuthInstanceType && len(instances.IDs()) > 0 && instances.checkIfExists(instance.GetID()) {
```

Comment:
Let's do the type assertion once at the top of this function, to avoid the multiple type assertions below

```suggestion
			if instance.Common().Type == CloudOAuthInstanceType && len(instances.IDs()) > 0 && instances.checkIfExists(instance.GetID()) {
			    cloudOAuthInstance, ok := instance.(*cloudOAuthInstance)
			    if !ok {
			        p.API.LogError("Instance type is `cloud-oauth` but failed to assert instance.(*cloudOAuthInstance).", "ID", instance.GetID())
			    } else {
			        // ...all other logic
```